azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/presentation/fragment/calling/lobby/ConnectingLobbyOverlayViewModel.kt

```diff
@@ -21,22 +22,31 @@ import com.azure.android.communication.ui.calling.redux.state.PermissionStatus
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 
-internal class ConnectingLobbyOverlayViewModel(private val dispatch: (Action) -> Unit) {
+internal class ConnectingLobbyOverlayViewModel(
+    private val dispatch: (Action) -> Unit,
+    private val isTelecomManagerEnabled: Boolean = false
+) {
 
     private lateinit var displayLobbyOverlayFlow: MutableStateFlow<Boolean>
     private lateinit var networkManager: NetworkManager
-
+    private lateinit var callingStatusStateFlow: MutableStateFlow<CallingStatus>
     private lateinit var cameraStateFlow: MutableStateFlow<CameraOperationalStatus>
     private lateinit var audioOperationalStatusStateFlow: MutableStateFlow<AudioOperationalStatus>
+    private var callType: CallType? = null
 
     fun getDisplayLobbyOverlayFlow(): StateFlow<Boolean> = displayLobbyOverlayFlow
 
+    fun getCallType(): CallType? = callType
+
+    fun getCallingStatusStateFlow(): StateFlow<CallingStatus> = callingStatusStateFlow
```

Comment:
Should these be visible to the View, or should it be distilled to view properties?

I.e. shouldDisplayLobbyOverlay is a function, but perhaps that should be the StateFlow and what the View is subscribing to, and it shouldn't know about callType/callStatus directly.

Reply:
As per discussion we will fix for GA.